crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -67,6 +73,36 @@ impl MatrixDriver {
         Ok(messages.chunk.into_iter().map(|ev| ev.event.cast()).collect())
     }
 
+    pub async fn read_state_events(
+        &self,
+        event_type: StateEventType,
+        state_key: &StateKeySelector,
+    ) -> Result<Vec<Raw<AnyTimelineEvent>>> {
+        let convert = |sync_or_stripped_state| match sync_or_stripped_state {
+            RawAnySyncOrStrippedState::Sync(ev) => Some(ev.cast()),
```

Comment:
Is this cast safe? We're casting from `AnySync...` (no jroom id) to one type that doesn't have `Sync` in the name (and thus should have a room ID), so trying to deserialize the latter may result in a deserialization error.

Reply:
Yeah, we even have a function for patching this on the other branch. Timo actually also noted this but I dismissed it because a bunch of other things to resolve in my head that seemed more important (it would have been caught sooner or later, I noted the same problem elsewhere previously). Thanks for showing me the exact location where we need to patch in the room ID. 🥲